### DISCUSSION TITLE
input.css is not applied to style.css (@layer styles)

### DISCUSSION THREAD

**Discussion by mahesh-riddles** [tailwindlabs/tailwindcss#11858] [Help] [answered]
`@layer components {
  .input-div {
    @apply ring-inset ring-1 border-2 rounded-md h-10 w-full sm:w-96 focus-within:ring-inset focus-within:ring-2 ring-purple-200 border-purple-200 focus-within:border-purple-400
  }
}`

`npx tailwindcss -i css/input.css -o css/style.css --watch`

`/** @type {import('tailwindcss').Config} */
module.exports = {
  content: ["./web/templates/**/*.{html,js}",],
  theme: {
    extend: {
    },
  },
  plugins: [
    require('@tailwindcss/forms'),
  ],
}
`

I can't find the ring-red-400 in style.css

**Comment by wongjn** [comment]
It seems like you're missing the `@tailwind` directives. Ensure you have the following somewhere in the input file:

```css
@tailwind base;
@tailwind components;
@tailwind utilities;
```

**Reply by wongjn** [reply, marked as answer]
> In the top, I have the following
> 
> ```
> @import url('https://fonts.googleapis.com/css?family=Roboto:300,400,500,700,900&display=swap');
> 
> @tailwind base;
> @tailwind components;
> @tailwind utilities;
> ```
> 
> and
> 
> ```
> 
> @layer components { .input-div { @apply ring-inset ring-1 border-2 rounded-md h-10 w-full sm:w-96 focus-within:ring-inset focus-within:ring-2 ring-purple-200 border-purple-200 focus-within:border-purple-400 } }
> 
> @layer components {
>   .input-div-error {
>     @apply ring-inset ring-1 border-2 rounded-md h-10 w-full sm:w-96 focus-within:ring-inset focus-within:ring-2 ring-red-400 border-red-200 focus-within:border-red-400
>   }
> }
> ```
> 
> If I am using any frontend framework like react, I would have created a component. But it is Golang. So spitting the code to component creates more work. It is easier to duplicate the input fields instead of making them as component in Golang. So I want to keep very very less style in input.css
> 
> But these styles are not populated in style.css from input.css[^1]

Consider checking `input-div` and `input-div-error` exist as unbroken strings in at least one file covered by the `content` file globs in the Tailwind configuration.

[^1]: https://github.com/tailwindlabs/tailwindcss/discussions/11858#discussioncomment-6771714

**Reply by mahesh-riddles** [reply]
aw... I understood now. They exist but I think tailwind can't read them. They are under if condition. Thanks for the help. I can fix it from here.

**Comment by mahesh-riddles** [comment]
In the top, I have the following

```
@import url('https://fonts.googleapis.com/css?family=Roboto:300,400,500,700,900&display=swap');

@tailwind base;
@tailwind components;
@tailwind utilities;
```

and
```

@layer components { .input-div { @apply ring-inset ring-1 border-2 rounded-md h-10 w-full sm:w-96 focus-within:ring-inset focus-within:ring-2 ring-purple-200 border-purple-200 focus-within:border-purple-400 } }

@layer components {
  .input-div-error {
    @apply ring-inset ring-1 border-2 rounded-md h-10 w-full sm:w-96 focus-within:ring-inset focus-within:ring-2 ring-red-400 border-red-200 focus-within:border-red-400
  }
}
```

If I am using any frontend framework like react, I would have created a component. But it is Golang. So spitting the code to component creates more work. It is easier to duplicate the input fields instead of making them as component in Golang. So I want to keep very very less style in input.css

But these styles are not populated in style.css from input.css

**Reply by wongjn** [reply]
Replied at https://github.com/tailwindlabs/tailwindcss/discussions/11858#discussioncomment-6771890 to keep the reply chain together.

**Comment by mahesh-riddles** [comment]
Am I doing anything wrong? Is it because of the some repeated styles in those two class?
